### PR TITLE
Feat: Add UI Unit Tests for Transfer Feature - 463

### LINKED ISSUE #463: LCFS - Comprehensive Unit Testing for Transfer Feature Status Logic and UI Validations
**Description:** The transfer feature within our application has evolved to include complex status logic and UI behaviors, particularly concerning button visibility and validation mechanisms. This task focuses on creating a suite of new unit tests to cover all aspects of this status logic, ensuring that the correct buttons render according to different statuses, validation processes function as intended, and specific UI elements like the 'sign and send', and 'sign and submit' buttons only appear when the signing declaration input is selected. These tests will help maintain the reliability and user experience of the transfer feature by preventing bugs and ensuring the application behaves as expected under various conditions.

**Purpose:** The aim of this task is to bolster the quality and robustness of the transfer feature by thoroughly testing its logic and UI elements. By implementing comprehensive unit tests, we aim to catch potential issues early in the development cycle, reduce the risk of regressions, and ensure a seamless experience for users. This will also facilitate future developments and enhancements by providing a solid testing foundation that verifies the application's behavior remains consistent as new changes are introduced.

**Acceptance Criteria:**
- [ ] New unit tests are created to cover all status logic associated with the transfer feature.
- [ ] Tests verify that the correct buttons are displayed for appropriate statuses, ensuring the UI aligns with the expected logic.
- [ ] Validation mechanisms are thoroughly tested to confirm they operate correctly under various scenarios.
- [ ] Tests ensure that the 'sign and send', and 'sign and submit' buttons are only visible when the signing declaration input is actively selected, and the user has the 'Signing Authority' role
- [ ] All new unit tests pass successfully and do not introduce any regressions in existing functionalities.
- [ ] Inline documentation is updated to include the new testing scenarios and the rationale behind each test case.

**Additional context:**
- Collaboration with the UI/UX team may be beneficial to clarify the intended behaviors and interactions within the transfer feature.
- Consider the impact of future changes to the transfer feature and design tests with extensibility in mind to accommodate potential updates without significant rework.

#455 Wait for the refactor of this component to be completed before writing these tests.

### PR DESCRIPTION
**Pull request by hamed-valiollahi**
This PR adds comprehensive UI unit tests for the transfer feature. Previously, there were 73 tests, and now there are 123 tests.

Closes #463

Screen Shot 2024-09-25 at 3 56 31 PM

### CODE REVIEW

File: frontend/src/views/Transfers/__tests__/buttonConfigs.test.jsx [hunk 1]
```diff
@@ -95,42 +92,69 @@ describe('buttonClusterConfigFn', () => {
...
-      const transferData = {
-        currentStatus: {
-          status: TRANSFER_STATUSES.DRAFT
-        },
-        agreementDate: new Date(),
-        transferHistory: [],
-        fromOrganization: {
+    }
+
+    it('should return deleteDraft, saveDraft, and signAndSend buttons', () => {
```
Comment (dhaselhan) [inline frontend/src/views/Transfers/__tests__/buttonConfigs.test.jsx:146]: question: The test mentions it tests for all 3 buttons, but in the asserts I only see a single .toBeDefined(), does one of them need to be updated?

File: frontend/src/views/Transfers/__tests__/buttonConfigs.test.jsx [hunk 2]
```diff
@@ -197,12 +210,216 @@ describe('buttonClusterConfigFn', () => {
...
+      transferHistory: [],
+      toOrganization: {
+        organizationId: 2
+      },
+      fromOrganization: {
+        organizationId: 1
+      }
+    }
+
     it('should return declineTransfer and signAndSubmit buttons for toOrganization user with roles', () => {
```
Comment (dhaselhan) [inline frontend/src/views/Transfers/__tests__/buttonConfigs.test.jsx:422]: Similar to above, looks like this is no longer testing for the declineTransfer button

File: frontend/src/views/Transfers/components/AttachmentList.jsx [hunk 3]
```diff
@@ -3,20 +3,20 @@ import BCBox from '@/components/BCBox'
...
       <Typography variant="h6" color="primary">
         Attachments
       </Typography>
       <List sx={{ maxWidth: '30%' }}>
-        {attachments.map((attachment) => (
+        {attachments.map((attachment, index) => (
           <ListItemButton
             component="a"
-            key={attachment.attachmentID}
+            key={`${attachment.attachmentID}-${index}`}
```
Comment (dhaselhan) [inline frontend/src/views/Transfers/components/AttachmentList.jsx:16]: Why is this necessary? Is the attachment ID not unique?

File: frontend/src/views/Transfers/components/TransferHistory.jsx [hunk 4]
```diff
@@ -39,13 +43,15 @@ function TransferHistory({ transferHistory }) {
     A: 6,
     B: 12
   }
-  const currentStatus = transferData?.currentStatus.status
-  const agreementDate = dayjs(transferData.agreementDate)
+  const currentStatus = transferData?.currentStatus?.status
```
Comment (dhaselhan) [inline frontend/src/views/Transfers/components/TransferHistory.jsx:46]: suggestion: Since you are checking transferData is defined above, you don't need to use ? here

File: frontend/src/views/Transfers/__tests__/buttonConfigs.test.jsx (hunk 1 above)
Comment (hamed-valiollahi) [reply frontend/src/views/Transfers/__tests__/buttonConfigs.test.jsx:146]: Good catch! I added assertions for all three buttons. 👍

File: frontend/src/views/Transfers/components/AttachmentList.jsx (hunk 3 above)
Comment (hamed-valiollahi) [reply frontend/src/views/Transfers/components/AttachmentList.jsx]: You're right. Initially, I thought it would be better to make attachments unique in case of duplicates, but it doesn't really make sense. I've reverted the changes. Thanks!

File: frontend/src/views/Transfers/components/TransferHistory.jsx (hunk 4 above)
Comment (hamed-valiollahi) [reply frontend/src/views/Transfers/components/TransferHistory.jsx]: Good point! I removed the `?` 👍

**Review by dhaselhan — APPROVED** [review]
Thanks for all the changes!
LGTM 👍
